framework/wazuh/engine/queries.py

```diff
@@ -0,0 +1,420 @@
+# Copyright (C) 2015, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from typing import Any, Dict, List, Type
+
+from wazuh.core.exception import WazuhError
+from wazuh.core.utils import filter_array_by_query
+
+
+class BaseQuery:
+    """Base class for response transformations."""
+
+    def apply_transformation(self, data: List[Dict[str, Any]]):
+        """Apply the transformation to the given data.
+
+        Args:
+            data: The data to be transformed.
+
+        Raises:
+            NotImplementedError: This method must be implemented in the derived classes.
+        """
+        raise NotImplementedError
+
+    @staticmethod
+    def can_i_run(params: Dict[str, Any], data: Any) -> bool:
+        """Check if the transformation can be applied to the given parameters and data.
+
+        Args:
+            params: The parameters used for the transformation.
+            data: The data to be transformed.
+
+        Returns:
+            bool: True if the transformation can be applied, False otherwise.
+
+        Raises:
+            NotImplementedError: This method must be implemented in the derived classes.
+        """
+        raise NotImplementedError
+
+
+class FieldSelector(BaseQuery):
+    """Transformation to select specific fields from the data."""
+
+    def __init__(self, params: Dict[str, Any]):
+        self.select = params['select']
+
+    @staticmethod
+    def can_i_run(params: Dict[str, Any], data: Any) -> bool:
+        if 'select' in params and params['select'] and isinstance(data, list):
+            return True
+        return False
+
+    def apply_transformation(self, data: List[Dict[str, Any]]):
```

Comment:
Transformation sounds like we are modifying the information but we are just filtering it. I think we should call the class `SelectFilter` and this method just `apply`.

Reply:
Changed to `apply` in commit https://github.com/wazuh/wazuh/commit/7f5d9c70cdd5191ed427fecce30a54885ce8ae4e
https://github.com/wazuh/wazuh/blob/92f0a22c401a9cdc766321dfd3229823ae849dbb/framework/wazuh/core/engine/queries.py#L11-L27